hedera-node/docs/design/services/smart-contract-service/evm-transaction-response-codes.md

```diff
@@ -0,0 +1,59 @@
+# EVM Transaction Response Codes in Hedera
+
+When executing an EVM transaction to a smart contract using `contractCall` in Hedera and it fails, a response code is present in the transaction receipt to indicate the top-level status.
+
+The default status for a failing `contractCall` is CONTRACT_REVERT_EXECUTED, which is included in the receipt. Child transactions might have more specific errors.
+In certain cases, the top-level status may differ from CONTRACT_REVERT_EXECUTED. The errors listed below are those special cases that can occur during the execution of a transaction and will be propagated as the top-level status.
+
+The handling of those cases currently happens [here](https://github.com/hashgraph/hedera-services/blob/774ed309600e4b4acd9e1ca72fcd87d354c8b9ff/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/hevm/HederaEvmTransactionResult.java#L141-L169), in the `HederaEvmTransactionResult` in the `finalStatus()` method.
```

Comment:
Code lines will change so I don't think it's necessary to note them. At the most you can do the `Class.function()` you have with a summarized mention of the cause.

Reply:
I agree. Will edit that.